Composer/packages/client/src/recoilModel/dispatchers/application.ts

```diff
@@ -157,6 +160,39 @@ export const applicationDispatcher = () => {
     await flushExistingTasks(callbackHelpers);
   });
 
+  const setSurveyEligibility = useRecoilCallback(({ set }: CallbackInterface) => () => {
+    const surveyStorage = new ClientStorage(window.localStorage, 'survey');
+
+    const optedOut = surveyStorage.get('optedOut', false);
+    if (optedOut) {
+      set(surveyEligibilityState, false);
+      return;
+    }
+
+    let days = surveyStorage.get('days', 0);
+    const lastUsed = surveyStorage.get('dateLastUsed', null);
+    const lastTaken = surveyStorage.get('epochLastTaken', null);
+    const today = new Date().toDateString();
+    if (lastUsed !== today) {
+      days += 1;
+      surveyStorage.set('days', days);
+    }
+    if (
+      // we need to have used Composer enough different calendar days to become eligible...
+      days >= SURVEY_PARAMETERS.daysUntilEligible &&
+      // ... and have either never taken the survey or it's been long enough since we did
+      (lastTaken == null || Date.now() - lastTaken > SURVEY_PARAMETERS.timeUntilNextSurvey)
+    ) {
+      // do we hit the probability of the survey appearing?
```

Comment:
Leave a better comment explaining what is happening here. It's not clear, and the comment is only vaguely helpful.